### PR TITLE
HPCC-30888 support ECLIDE build with github action

### PR DESCRIPTION
**Pull request by xwang2713**
Also add code sign for Windows builds

Since this is the first time I edit github action workflow some code may not 100% following our guideline.
Hopefully Gordon and Michael can help me review it.

The test will require having encoded code sign certificate. I tested with following encoding:
 [convert]::ToBase64String((Get-Content -path "hpcc_code_signing.pfx" -Encoding byte))

I can't make base64 encode/decode work with github action

I verify the yaml with yamllint
yamllint -d "{extends: relaxed, rules: {line-length: disable}}"  build-assets.yml
Removed some end spaces according to above output

signed off by xiaoming.wang@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

I did test the code in a separated private branch
I will ask Michael to test it

### CODE REVIEW

**Comment by xwang2713** [comment]
I assigned JIRA https://track.hpccsystems.com/browse/HPCC-30888 to my name, xwang2713

File: .github/workflows/build-assets.yml [hunk 1]
```diff
@@ -484,3 +516,121 @@ jobs:
...
+          cd ./ECLIDE/build
+          C:\"Program Files (x86)"\"Microsoft SDKs"\ClickOnce\SignTool\signtool.exe sign /debug /f ../../sign/hpcc_code_signing.pfx /p $env:SIGN_PW /t http://timestamp.digicert.com /fd SHA256 hpccsystems-eclide*.exe
+
+      - name: Upload Assets
+        uses: ncipollo/release-action@v1.12.0
+        with:
+          allowUpdates: true
+          generateReleaseNotes: false
+          prerelease: ${{ contains(github.ref, '-rc') }}
+          artifacts: "./ECLIDE/build/*.exe,./build/*.exe,./build/*.msi,./build/*.dmg,./build/*.pkg,./build/*.tar.gz"
```
Comment (GordonSmith) [inline .github/workflows/build-assets.yml:629]: Just `./ECLIDE/build/*.exe` should be needed?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: .github/workflows/build-assets.yml [hunk 2]
```diff
@@ -427,6 +427,22 @@ jobs:
           submodules: recursive
           path: ./LN
 
+      - name: "Create code sining"
```
[the remark this answers is not in the document]
Comment (xwang2713) [reply .github/workflows/build-assets.yml:430]: Fixed

File: .github/workflows/build-assets.yml (hunk 1 above)
Comment (xwang2713) [reply .github/workflows/build-assets.yml:629]: Fixed

**Review by GordonSmith — APPROVED** [review]

**Review by Michael-Gardner — APPROVED** [review]
PR looks solid. I made some changes that Ming took to his branch regarding eclide builds and upload-artifact / download-artifact instead of curl.  It's now running for users who are not hpcc-systems.

For SIGNING_CERTIFICATE I was able to get it to work with a cert converted to the pks12 format and then encoded using the linux base64 utility.  The cert will likely need to be added to our secrets.

Successful run of this commit is here: https://github.com/Michael-Gardner/HPCC-Platform/actions/runs/7627989492

Artifacts present here: https://github.com/Michael-Gardner/HPCC-Platform/releases/tag/community_ming-30888-pr1

**Comment by Michael-Gardner** [comment]
@ghalliday should now be ready for a final look

**Pull request merged**